### PR TITLE
Add SVG sanitization and security hardening for PlantUML diagrams

### PR DESCRIPTION
**Pull request by michael72**
## Summary
This PR adds comprehensive security hardening for PlantUML diagram rendering by implementing SVG sanitization, Content Security Policy headers, and safer redirect handling. Since the PlantUML server URL is user/workspace-configurable, the SVG responses must be treated as untrusted input.

## Key Changes

- **SVG Sanitization** (`svgSanitizer.ts`): New module that strips active content from SVG responses including:
  - Dangerous elements: `<script>`, `<foreignObject>`, `<iframe>`, `<object>`, `<embed>`, `<form>`, `<meta>`, `<base>`, `<link>`
  - Event handler attributes (`onclick`, `onload`, etc.)
  - JavaScript URLs (`javascript:` scheme in href/src attributes)
  - Non-image data URLs (preserves legitimate `data:image/*` URIs)
  - Handles nested/split tags through iterative stripping

- **Redirect Security** (`plantumlService.ts`): 
  - Added `resolveRedirectUrl()` to validate and resolve redirect locations
  - Rejects redirects that downgrade HTTPS to HTTP
  - Properly resolves relative redirect paths against the server URL
  - Integrated sanitization into the main `fetchSvg()` function

- **Content Security Policy** (`previewPanel.ts`):
  - Added CSP header (`default-src 'none'; style-src 'unsafe-inline'`) to both preview and error pages
  - Improved nonce generation using cryptographically secure `crypto.randomBytes()` instead of pseudo-random string generation
  - Enhanced HTML escaping to include quotes for attribute context

- **Code Quality** (`pumlsrvService.ts`):
  - Changed `execSync("which pumlsrv")` to `execFileSync("which", ["pumlsrv"])` to avoid shell spawning

## Testing
Comprehensive test suite added for `sanitizeSvg()` covering:
- Preservation of legitimate PlantUML SVG content
- Removal of various attack vectors
- Edge cases like unclosed/self-closing dangerous tags
- Nested and split tag handling

Additional integration tests verify sanitization in the fetch pipeline and redirect security.

## Defense-in-Depth
The webview CSP is the primary defense against script execution; the SVG sanitizer provides defense-in-depth by removing active content that has no legitimate use in diagram rendering.

https://claude.ai/code/session_01L9e4fDgZ6dRqgRq8Z8agj9

### CODE REVIEW

File: src/plantumlService.ts [hunk 1]
```diff
@@ -33,16 +34,42 @@ export function getRenderMethod(): "get" | "post" {
...
+/**
+ * Resolves a redirect Location header against the URL of the original
+ * request. Relative locations are resolved; redirects that would
+ * downgrade an HTTPS request to plain HTTP (or switch to a non-HTTP
+ * scheme) are rejected.
+ */
+function resolveRedirectUrl(location: string, baseUrl: string): string {
+  const resolved = new URL(location, baseUrl);
+  if (resolved.protocol !== "https:" && resolved.protocol !== "http:") {
+    throw new Error(
```
Comment (michael72) [inline src/plantumlService.ts:60]: this line is not tested

File: src/plantumlService.ts [hunk 2]
```diff
@@ -150,7 +183,13 @@ async function fetchSvgViaPost(
       const statusCode = response.statusCode ?? 0;
       const location = response.headers.location ?? "";
       if (statusCode >= 300 && statusCode < 400 && location.length > 0) {
-        fetchFromUrl(location).then(resolve).catch(reject);
+        try {
+          fetchFromUrl(resolveRedirectUrl(location, postUrl))
+            .then(resolve)
+            .catch(reject);
+        } catch (error) {
+          reject(error instanceof Error ? error : new Error(String(error)));
```
Comment (michael72) [inline src/plantumlService.ts:193]: this line is not tested

**Review by michael72 — COMMENTED** [review]
please add tests for the untested lines

File: src/plantumlService.ts (hunk 1 above)
Comment (michael72) [reply src/plantumlService.ts:60]: Covered in cf15c79 by the new test "should reject redirects to non-HTTP URL schemes" (redirect to `file:///etc/passwd` now exercises this throw).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/plantumlService.ts (hunk 2 above)
Comment (michael72) [reply src/plantumlService.ts:193]: Covered in cf15c79 by the new test "should reject invalid redirects in POST mode". The defensive non-Error fallback in the ternary is unreachable (resolveRedirectUrl only throws Error), so it's now marked with a v8 ignore annotation.

---
_Generated by [Claude Code](https://claude.ai/code)_
